server/src/postgresql/handler.rs

```diff
@@ -0,0 +1,189 @@
+// Copyright 2023 The CeresDB Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{sync::Arc, time::Duration};
+
+use async_trait::async_trait;
+use common_types::datum::{Datum, DatumKind};
+use datafusion::parquet::data_type::AsBytes;
+use futures::stream;
+use interpreters::interpreter::Output;
+use log::error;
+use pgwire::{
+    api::{
+        query::SimpleQueryHandler,
+        results::{DataRowEncoder, FieldFormat, FieldInfo, QueryResponse, Response, Tag},
+        ClientInfo, Type,
+    },
+    error::{PgWireError, PgWireResult},
+};
+use proxy::{context::RequestContext, http::sql::Request, Proxy};
+use query_engine::executor::Executor as QueryExecutor;
+use snafu::ResultExt;
+
+use crate::postgresql::error::{CreateContext, Result};
+pub struct PostgresqlHandler<Q> {
+    pub(crate) proxy: Arc<Proxy<Q>>,
+    pub(crate) timeout: Option<Duration>,
+}
+
+#[async_trait]
+impl<Q: QueryExecutor + 'static> SimpleQueryHandler for PostgresqlHandler<Q> {
+    async fn do_query<'a, C>(&self, _client: &C, sql: &'a str) -> PgWireResult<Vec<Response<'a>>>
+    where
+        C: ClientInfo + Unpin + Send + Sync,
+    {
+        let ctx = self.create_ctx().map_err(|e| {
+            PgWireError::UserError(Box::new(pgwire::error::ErrorInfo::new(
+                "ERROR".to_owned(),
+                "08000".to_owned(),
```

Comment:
What does this mean? better to add some comments.

Reply:
I'll change this to `PgWireError::ApiError(Box::new(e))`